test/unit/visitor/semantic_analysis.cpp

```diff
@@ -192,3 +192,21 @@ SCENARIO("At most one DERIVATIVE block", "[visitor][semantic_analysis]") {
         }
     }
 }
+
+SCENARIO("Recursive function calls", "[visitor][semantic_analysis]") {
+    GIVEN("A file with recursive function calls") {
+        std::string nmodl_text = R"(
+            FUNCTION a() {
+                b()
+                a = 42
+            }
+            PROCEDURE b() {
+                LOCAL x
+                x = a()
+            }
+        )";
+        THEN("Semantic analysis should fail") {
+            REQUIRE(run_semantic_analysis_visitor(nmodl_text));
+        }
```

Comment:
This reads like the analysis succeeded on something that contains an infinite loop. Shouldn't there be a `REQUIRE_THROWS` or a `!` something to make it assert failure?

Reply:
Thank you!